pkg/forwarder/interfaces.go

```diff
@@ -25,17 +25,18 @@ type VertexBuffer struct {
 // ToWhichStepDecider decides which step to forward after applying the WhereTo function.
 type ToWhichStepDecider interface {
 	// WhereTo decides where to forward the result to based on the name of the step it returns.
+	// It is deterministic and will forward the same message-id to the partition using a hashing algorithm.
 	// It supports 2 addition keywords which need not be a step name. They are "ALL" and "DROP"
 	// where former means, forward to all the neighbouring steps and latter means do not forward anywhere.
-	WhereTo([]string, []string) ([]VertexBuffer, error)
+	WhereTo([]string, []string, string) ([]VertexBuffer, error)
```

Comment:
Can we rewrite the docs above, and explain the args accepted? I feel like the docs have drifted too much, and no one understands what they tell.